catalog/spatial/csw/spatial-csw-source-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/source/reader/GetRecordsMessageBodyReader.java

```diff
@@ -98,6 +103,14 @@ public CswRecordCollection readFrom(Class<CswRecordCollection> type, Type generi
             throws IOException, WebApplicationException {
 
         CswRecordCollection cswRecords = null;
+        Map<String, Serializable> resourceProperties = new HashMap<>();
+        // Check if the server returned a Partial Content response (hopefully in response to a range header)
+        String contentRangeHeader = httpHeaders.getFirst(HttpHeaders.CONTENT_RANGE);
+        if (StringUtils.isNotBlank(contentRangeHeader)) {
+            contentRangeHeader = StringUtils.substringBetween(contentRangeHeader.toLowerCase(), "bytes ", "-");
```

Comment:
Same regarding duplicate code.

Might be worth creating a small service for content range processing with two methods on it: one for retrieving the `bytesSkipped` and one for aligning input streams.